context/context.go

```diff
@@ -0,0 +1,20 @@
+package context
+
+import (
+	gocontext "context"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// TODO:
+type Context interface {
+	WithContext(ctx gocontext.Context) Context
+	WithTimeout(timeout time.Duration) (Context, func())
+
+	Errorf(format string, args ...any)
+
+	StartTrace(tracerName string, spanName string) (Context, trace.Span)
```

Comment:
Can we not have the span as a field on the context? 

and then `ctx.SpanAttribute(key, value)` or `ctx.SpanAttributes(key...)`

Reply:
I've gone with a separate approach. About to push.